### PR TITLE
Trigger errors for unresolvable parameters

### PR DESCRIPTION
**Pull request by rvanvelzen**

### CODE REVIEW

File: src/Rules/FunctionCallParametersCheck.php [hunk 1]
```diff
@@ -268,6 +268,18 @@ public function check(
 				))->line($argumentLine)->build();
 			}
 
+			if ($this->unresolvableTypeHelper->containsUnresolvableType($parameterType)) {
```
Comment (ondrejmirtes) [inline src/Rules/FunctionCallParametersCheck.php]: These contributions make me drool! So good 😊 But it should be reported only with bleedingEdge enabled (by adding new featureToggles parameter), who knows what's it gonna unleash in real codebases 😊

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

**Comment by rvanvelzen** [comment]
The one test failure is a BC break for a non-`@api` method call. Should I make the message optional, or is the breakage acceptable?

**Comment by canvural** [comment]
@rvanvelzen I think it's fine. It's my fault to use non-`@api` calls 😄  I'll fix it there.

**Comment by ondrejmirtes** [comment]
@canvural Maybe I suggested to you myself to use FunctionCallParametersCheck so I'm sorry about it :) There should be a way in Larastan to do it nicer, just an idea: allow custom rule developer to invoke their own virtual nodes. You could invoke a custom `FuncCall` node and the rules in PHPStan would pick it up themselves.

But for now, feel free to prepare a new Larastan release that requires PHPStan ^1.7 (after this PR is merged), we can then coordinate the release together :)

**Comment by ondrejmirtes** [comment]
Or pass the extra argument to FunctionCallParametersCheck, ignore PHPStan error for now, release it (and not change the PHPStan requirements), and clean it up after 1.7 is releaased :)

**Comment by ondrejmirtes** [comment]
@rvanvelzen Just one more thing - have you looked into the failures before the new check was behind a feature toggle?

* https://github.com/phpstan/phpstan-src/runs/6453862910?check_suite_focus=true
* https://github.com/phpstan/phpstan-src/runs/6453863040?check_suite_focus=true
* https://github.com/phpstan/phpstan-src/runs/6453863215?check_suite_focus=true
* https://github.com/phpstan/phpstan-src/runs/6453863498?check_suite_focus=true

It'd be good to fix them now since it's a valuable piece of feedback about the new check :)

**Comment by rvanvelzen** [comment]
@ondrejmirtes yeah, I fixed most of them in f3c5928. There's one left that I'll try to fix.

**Comment by ondrejmirtes** [comment]
Thank you!

**Comment by ondrejmirtes** [comment]
I went the optional way as that avoids all of the headaches: https://github.com/phpstan/phpstan-src/commit/a114a3f1c4158723786aadaf67d9abcf6b0e0b88

**Pull request merged**